### PR TITLE
optparse: fix segfault when subcommand registration fails due to invalid options table

### LINKED ISSUE #5731: liboptparse segfaults with duplicate subcommand option table entries
Problem: when a subcommand option table contains a duplicate entry, a segfault occurs during registration.  Although this is triggered by incorrect code,  the handling of it could be a tad more user friendly.  Note this seems to be specific to sub-commands. 

To recreate
```diff
diff --git a/src/cmd/flux-module.c b/src/cmd/flux-module.c
index 59294f80b..aa7c760ac 100644
--- a/src/cmd/flux-module.c
+++ b/src/cmd/flux-module.c
@@ -49,6 +49,9 @@ static struct optparse_option list_opts[] = {
 };
 
 static struct optparse_option remove_opts[] =  {
+    { .name = "force", .key = 'f', .has_arg = 0,
+      .usage = "Ignore nonexistent modules",
+    },
     { .name = "force", .key = 'f', .has_arg = 0,
       .usage = "Ignore nonexistent modules",
     },
```
```
$ ./flux module
Segmentation fault (core dumped)
$ coredumpctl debug
[snip]
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007b14026a53fe in __GI___libc_free (mem=0x5d310cd22)
    at ./malloc/malloc.c:3368
3368	./malloc/malloc.c: No such file or directory.
(gdb) bt
#0  0x00007b14026a53fe in __GI___libc_free (mem=0x5d310cd22)
    at ./malloc/malloc.c:3368
#1  0x00007b14029b8b0c in optparse_destroy (p=p@entry=0x5d310cd229d0)
    at optparse.c:702
#2  0x00007b14029b9c21 in optparse_reg_subcommand (
    opts=0x5d310c696500 <remove_opts>, flags=0, 
    doc=0x5d310c693313 "Unload module", 
    usage=0x5d310c693302 "[OPTIONS] module", cb=<optimized out>, 
    name=<optimized out>, p=<optimized out>) at optparse.c:815
#3  optparse_reg_subcommand (p=<optimized out>, name=<optimized out>, 
    cb=<optimized out>, usage=0x5d310c693302 "[OPTIONS] module", 
    doc=0x5d310c693313 "Unload module", flags=0, 
    opts=0x5d310c696500 <remove_opts>) at optparse.c:793
#4  0x00007b14029b9c9c in optparse_reg_subcommands (p=p@entry=0x5d310cd202a0, 
    cmds=cmds@entry=0x5d310c696020 <subcommands>) at optparse.c:831
#5  0x00005d310c690973 in main (argc=1, argv=0x7ffd8460f838)
    at flux-module.c:193
```

### PR DESCRIPTION
**Pull request by grondo**
This PR fixes #5731. The segfault is actually caused when destroying an optparse child object (subcommand) before the parent. In this case `optparse_destroy()` is called recursively and it isn't re-entrant safe. This PR fixes that after cleaning up some of the code formatting. A test is added to the unit tests as well.

### CODE REVIEW

File: src/common/liboptparse/optparse.h [hunk 1]
```diff
@@ -37,8 +37,9 @@ typedef int (*opt_fatalerr_f) (void *h, int exit_code);
 /*
  *  prototype for option callback hook
  */
-typedef int (*optparse_cb_f) (optparse_t *p, struct optparse_option *o,
-			      const char *optarg);
+typedef int (*optparse_cb_f) (optparse_t *p,
+                              struct optparse_option *o,
+			                  const char *optarg);
```
Comment (chu11) [inline src/common/liboptparse/optparse.h:42]: is last line of this indent off?

**Review by chu11 — APPROVED** [review]
LGTM, just one indentation looked off to me (hopefully not just my browser)

**Review by garlick — APPROVED** [review]
LGTM!

File: src/common/liboptparse/optparse.h (hunk 1 above)
Comment (grondo) [reply src/common/liboptparse/optparse.h]: Some tabs snuck in there! Fixed.

**Comment by grondo** [comment]
Thanks! I've set MWP.

**Pull request merged**